feat(compiler): drop support for TypeScript 4.8

crisbeto opened the pull request:
Drops support for TypeScript 4.8 from the compiler and removes all of the compatibility code we had for it.

BREAKING CHANGE:
* TypeScript 4.8 is no longer supported.

josephperrott reviewed (approved):
LGTM

JoostK reviewed (approved):
Nice! Do we want a "BREAKING CHANGES" note for this one?

crisbeto committed:
feat(compiler): drop support for TypeScript 4.8

Drops support for TypeScript 4.8 from the compiler and removes all of the compatibility code we had for it.

BREAKING CHANGE:
* TypeScript 4.8 is no longer supported.

AndrewKushnir commented:
This PR was merged into the repository by commit 79cdfeb3921687dfbc8fea8d9f7ba4dbb14a7193.
